### ISSUE TITLE
Add support for NTP version 1

### ISSUE THREAD

**Issue by mdavids** [pendulum-project/ntpd-rs#734]
NTP version 1 still occurs in the wild (for IPv4, not so much for IPv6), but ntpd-rs (as a server) does not seem to support version 1 (RFC1059) given the fact it logs these messages: `received invalid packet: Invalid version 1` .

Should ntpd-rs get NTP version 1 support?

BTW: a heavily loaded server will soon run out of disk space when every refused packet generates a log line. Is this tunable?

**Comment by CreeperFace00** [comment]
Are there any plans to add v1 support? I do feel a little bad not offering v1 to clients that need it.

**Comment by davidv1992** [comment]
We are currently discussing internally what we want to do with this.

**Comment by davidv1992** [comment]
From a server we have statistics for, current use distribution is roughly:
```
Query distribution:
NTPv1:   339,935
NTPv2:     5,282
NTPv3: 5,993,413
NTPv4: 5,213,351
```

**Comment by mlichvar** [comment]
Most of those, at least the last time I looked at it, were not real NTPv1 clients. The NTPv1 header has a reserved field instead of the mode (which should be determined from the source port number), but these clients set it to 3. Also, the NTPv1 header has different fields in place of the NTPv2+ root delay and dispersion.

So the question is if you want to support these broken clients and/or the proper NTPv1. A full support of NTPv1 is dangerous due to the possibility of a NAT breaking the mode detection (server misinterpreted as client) and causing a flood of NTP traffic. 

There was a similar request for ntpsec: https://gitlab.com/NTPsec/ntpsec/-/issues/707

chronyd added partial support to provide similar functionality as ntpd and make the users happy, although it would be better if all implementations dropped it and the clients were fixed.
